{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Populating the interactive namespace from numpy and matplotlib\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "%pylab inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "import drugz as dz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You can run functions individually"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>T0</th>\n",
       "      <th>T15_A_control</th>\n",
       "      <th>T15_B_control</th>\n",
       "      <th>T15_C_control</th>\n",
       "      <th>T15_A_olaparib</th>\n",
       "      <th>T15_B_olaparib</th>\n",
       "      <th>T15_C_olaparib</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sgRNA</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG_CACCTTCGAGCTGCTGCGCG</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>313</td>\n",
       "      <td>235</td>\n",
       "      <td>47</td>\n",
       "      <td>337</td>\n",
       "      <td>428</td>\n",
       "      <td>115</td>\n",
       "      <td>340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_AAGAGCGCCTCGGTCCCAGC</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>99</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_TGGACTTCCAGCTACGGCGC</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>650</td>\n",
       "      <td>336</td>\n",
       "      <td>74</td>\n",
       "      <td>185</td>\n",
       "      <td>392</td>\n",
       "      <td>193</td>\n",
       "      <td>304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_CACTGGCGCCATCGAGAGCC</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>718</td>\n",
       "      <td>192</td>\n",
       "      <td>34</td>\n",
       "      <td>296</td>\n",
       "      <td>178</td>\n",
       "      <td>69</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_GCTCGGGCTTGTCCACAGGA</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>180</td>\n",
       "      <td>230</td>\n",
       "      <td>29</td>\n",
       "      <td>122</td>\n",
       "      <td>394</td>\n",
       "      <td>148</td>\n",
       "      <td>364</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Gene   T0  T15_A_control  T15_B_control  \\\n",
       "sgRNA                                                                \n",
       "A1BG_CACCTTCGAGCTGCTGCGCG  A1BG  313            235             47   \n",
       "A1BG_AAGAGCGCCTCGGTCCCAGC  A1BG   99              8              1   \n",
       "A1BG_TGGACTTCCAGCTACGGCGC  A1BG  650            336             74   \n",
       "A1BG_CACTGGCGCCATCGAGAGCC  A1BG  718            192             34   \n",
       "A1BG_GCTCGGGCTTGTCCACAGGA  A1BG  180            230             29   \n",
       "\n",
       "                           T15_C_control  T15_A_olaparib  T15_B_olaparib  \\\n",
       "sgRNA                                                                      \n",
       "A1BG_CACCTTCGAGCTGCTGCGCG            337             428             115   \n",
       "A1BG_AAGAGCGCCTCGGTCCCAGC             13              26               5   \n",
       "A1BG_TGGACTTCCAGCTACGGCGC            185             392             193   \n",
       "A1BG_CACTGGCGCCATCGAGAGCC            296             178              69   \n",
       "A1BG_GCTCGGGCTTGTCCACAGGA            122             394             148   \n",
       "\n",
       "                           T15_C_olaparib  \n",
       "sgRNA                                      \n",
       "A1BG_CACCTTCGAGCTGCTGCGCG             340  \n",
       "A1BG_AAGAGCGCCTCGGTCCCAGC              28  \n",
       "A1BG_TGGACTTCCAGCTACGGCGC             304  \n",
       "A1BG_CACTGGCGCCATCGAGAGCC             185  \n",
       "A1BG_GCTCGGGCTTGTCCACAGGA             364  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# import test dataset\n",
    "\n",
    "# func load_reads(filepath, index_column, genes_to_remove)\n",
    "data = dz.load_reads(\n",
    "    \"./sgRNA_count.txt\", index_column=0, genes_to_remove=[\"LacZ, luciferase, EGFR\"]\n",
    ")\n",
    "data.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>T15_A_olaparib</th>\n",
       "      <th>T15_B_olaparib</th>\n",
       "      <th>T15_C_olaparib</th>\n",
       "      <th>T15_A_control</th>\n",
       "      <th>T15_B_control</th>\n",
       "      <th>T15_C_control</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sgRNA</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG_CACCTTCGAGCTGCTGCGCG</th>\n",
       "      <td>1.610233e+06</td>\n",
       "      <td>7.855191e+05</td>\n",
       "      <td>1.377076e+06</td>\n",
       "      <td>1.172070e+06</td>\n",
       "      <td>9.437751e+05</td>\n",
       "      <td>1.787798e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_AAGAGCGCCTCGGTCCCAGC</th>\n",
       "      <td>9.781791e+04</td>\n",
       "      <td>3.415301e+04</td>\n",
       "      <td>1.134062e+05</td>\n",
       "      <td>3.990025e+04</td>\n",
       "      <td>2.008032e+04</td>\n",
       "      <td>6.896552e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_TGGACTTCCAGCTACGGCGC</th>\n",
       "      <td>1.474793e+06</td>\n",
       "      <td>1.318306e+06</td>\n",
       "      <td>1.231268e+06</td>\n",
       "      <td>1.675810e+06</td>\n",
       "      <td>1.485944e+06</td>\n",
       "      <td>9.814324e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_CACTGGCGCCATCGAGAGCC</th>\n",
       "      <td>6.696764e+05</td>\n",
       "      <td>4.713115e+05</td>\n",
       "      <td>7.492912e+05</td>\n",
       "      <td>9.576060e+05</td>\n",
       "      <td>6.827309e+05</td>\n",
       "      <td>1.570292e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_GCTCGGGCTTGTCCACAGGA</th>\n",
       "      <td>1.482318e+06</td>\n",
       "      <td>1.010929e+06</td>\n",
       "      <td>1.474281e+06</td>\n",
       "      <td>1.147132e+06</td>\n",
       "      <td>5.823293e+05</td>\n",
       "      <td>6.472149e+05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           T15_A_olaparib  T15_B_olaparib  T15_C_olaparib  \\\n",
       "sgRNA                                                                       \n",
       "A1BG_CACCTTCGAGCTGCTGCGCG    1.610233e+06    7.855191e+05    1.377076e+06   \n",
       "A1BG_AAGAGCGCCTCGGTCCCAGC    9.781791e+04    3.415301e+04    1.134062e+05   \n",
       "A1BG_TGGACTTCCAGCTACGGCGC    1.474793e+06    1.318306e+06    1.231268e+06   \n",
       "A1BG_CACTGGCGCCATCGAGAGCC    6.696764e+05    4.713115e+05    7.492912e+05   \n",
       "A1BG_GCTCGGGCTTGTCCACAGGA    1.482318e+06    1.010929e+06    1.474281e+06   \n",
       "\n",
       "                           T15_A_control  T15_B_control  T15_C_control  \n",
       "sgRNA                                                                   \n",
       "A1BG_CACCTTCGAGCTGCTGCGCG   1.172070e+06   9.437751e+05   1.787798e+06  \n",
       "A1BG_AAGAGCGCCTCGGTCCCAGC   3.990025e+04   2.008032e+04   6.896552e+04  \n",
       "A1BG_TGGACTTCCAGCTACGGCGC   1.675810e+06   1.485944e+06   9.814324e+05  \n",
       "A1BG_CACTGGCGCCATCGAGAGCC   9.576060e+05   6.827309e+05   1.570292e+06  \n",
       "A1BG_GCTCGGGCTTGTCCACAGGA   1.147132e+06   5.823293e+05   6.472149e+05  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# normalize readcounts\n",
    "\n",
    "# func normalize_readconts(reads, control_samples, treatment_samples)\n",
    "norm_data = dz.normalize_readcounts(\n",
    "    data,\n",
    "    [\"T15_A_control\", \"T15_B_control\", \"T15_C_control\"],\n",
    "    [\"T15_A_olaparib\", \"T15_B_olaparib\", \"T15_C_olaparib\"],\n",
    ")\n",
    "norm_data.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>T15_A_control</th>\n",
       "      <th>eb_std_0</th>\n",
       "      <th>zscore_fc_0</th>\n",
       "      <th>fc_0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1CF_CGTGGCTATTTGGCATACAC</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>452</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.034625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_TGGACTTCCAGCTACGGCGC</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.184346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_CAAGAGAAAGACCACGAGCA</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>300</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.119857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_CACCTTCGAGCTGCTGCGCG</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>235</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.458209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_GCTCGGGCTTGTCCACAGGA</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>230</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.369821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_CACTGGCGCCATCGAGAGCC</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>192</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.515965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF_GACATGGTATTGCAGTAGAC</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>183</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.329980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF_GGTATACTCTCCTTGCAGCA</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>69</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.069698</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG_AAGAGCGCCTCGGTCCCAGC</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>8</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.293594</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           GENE  T15_A_control  eb_std_0  zscore_fc_0  \\\n",
       "A1CF_CGTGGCTATTTGGCATACAC  A1CF            452       0.0          0.0   \n",
       "A1BG_TGGACTTCCAGCTACGGCGC  A1BG            336       0.0          0.0   \n",
       "A1BG_CAAGAGAAAGACCACGAGCA  A1BG            300       0.0          0.0   \n",
       "A1BG_CACCTTCGAGCTGCTGCGCG  A1BG            235       0.0          0.0   \n",
       "A1BG_GCTCGGGCTTGTCCACAGGA  A1BG            230       0.0          0.0   \n",
       "A1BG_CACTGGCGCCATCGAGAGCC  A1BG            192       0.0          0.0   \n",
       "A1CF_GACATGGTATTGCAGTAGAC  A1CF            183       0.0          0.0   \n",
       "A1CF_GGTATACTCTCCTTGCAGCA  A1CF             69       0.0          0.0   \n",
       "A1BG_AAGAGCGCCTCGGTCCCAGC  A1BG              8       0.0          0.0   \n",
       "\n",
       "                               fc_0  \n",
       "A1CF_CGTGGCTATTTGGCATACAC -0.034625  \n",
       "A1BG_TGGACTTCCAGCTACGGCGC -0.184346  \n",
       "A1BG_CAAGAGAAAGACCACGAGCA -0.119857  \n",
       "A1BG_CACCTTCGAGCTGCTGCGCG  0.458209  \n",
       "A1BG_GCTCGGGCTTGTCCACAGGA  0.369821  \n",
       "A1BG_CACTGGCGCCATCGAGAGCC -0.515965  \n",
       "A1CF_GACATGGTATTGCAGTAGAC -0.329980  \n",
       "A1CF_GGTATACTCTCCTTGCAGCA  0.069698  \n",
       "A1BG_AAGAGCGCCTCGGTCCCAGC  1.293594  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate fold change\n",
    "\n",
    "# func calculate_fold_change(reads, normalized_counts, control_samples, treatment_samples, pseudocount, replicate):\n",
    "\n",
    "lfc = []\n",
    "# for each replicate\n",
    "for i in range(2):\n",
    "    lfc.append(\n",
    "        dz.calculate_fold_change(\n",
    "            data,\n",
    "            norm_data,\n",
    "            [\"T15_A_control\", \"T15_B_control\", \"T15_C_control\"],\n",
    "            [\"T15_A_olaparib\", \"T15_B_olaparib\", \"T15_C_olaparib\"],\n",
    "            5,\n",
    "            i,\n",
    "        )\n",
    "    )\n",
    "lfc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(                           GENE  T15_A_control  eb_std_0  zscore_fc_0  \\\n",
       " A1CF_CGTGGCTATTTGGCATACAC  A1CF            452  0.540492    -0.064062   \n",
       " A1BG_TGGACTTCCAGCTACGGCGC  A1BG            336  0.540492    -0.341071   \n",
       " A1BG_CAAGAGAAAGACCACGAGCA  A1BG            300  0.540492    -0.221756   \n",
       " A1BG_CACCTTCGAGCTGCTGCGCG  A1BG            235  0.540492     0.847764   \n",
       " A1BG_GCTCGGGCTTGTCCACAGGA  A1BG            230  0.540492     0.684232   \n",
       " A1BG_CACTGGCGCCATCGAGAGCC  A1BG            192  0.540492    -0.954621   \n",
       " A1CF_GACATGGTATTGCAGTAGAC  A1CF            183  0.540492    -0.610518   \n",
       " A1CF_GGTATACTCTCCTTGCAGCA  A1CF             69  0.540492     0.128953   \n",
       " A1BG_AAGAGCGCCTCGGTCCCAGC  A1BG              8  0.540492     2.393365   \n",
       " \n",
       "                                fc_0  \n",
       " A1CF_CGTGGCTATTTGGCATACAC -0.034625  \n",
       " A1BG_TGGACTTCCAGCTACGGCGC -0.184346  \n",
       " A1BG_CAAGAGAAAGACCACGAGCA -0.119857  \n",
       " A1BG_CACCTTCGAGCTGCTGCGCG  0.458209  \n",
       " A1BG_GCTCGGGCTTGTCCACAGGA  0.369821  \n",
       " A1BG_CACTGGCGCCATCGAGAGCC -0.515965  \n",
       " A1CF_GACATGGTATTGCAGTAGAC -0.329980  \n",
       " A1CF_GGTATACTCTCCTTGCAGCA  0.069698  \n",
       " A1BG_AAGAGCGCCTCGGTCCCAGC  1.293594  ,\n",
       " 'zscore_fc_0')"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate the smoothed variance\n",
    "\n",
    "# func empirical_bayes(fold_change, half_window_size, no_of_guides, fc_replicate_id,\n",
    "# empirical_bayes_id, fc_zscore_id)\n",
    "for i in range(len(lfc)):\n",
    "    lfc[i] = dz.empirical_bayes(\n",
    "        lfc[i], 5, 9, \"fc_\" + str(i), \"eb_std_\" + str(i), \"zscore_fc_\" + str(i)\n",
    "    )\n",
    "lfc[0]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "or, you can pass the arguments and run the whole analysis at once"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# define the Arguements class (more convinient since iPython doesn't recognize argparse arguments)\n",
    "# these are user-specified arguments\n",
    "\n",
    "# infile = input readcounts matrix\n",
    "# drugz_out_file = name of a file in which you will write the drugz results\n",
    "# control_samples = the names of control samples (included in column names)\n",
    "# drug_samples = the names of drug-treated samples (included in column names)\n",
    "# unpaired = unpaired approach - compares mean(treated samples) to mean(control samples)\n",
    "# pseudocount = counts added to the observed readscounts, default = 5\n",
    "# half_window_size = size of the first bin and half the size of the inital sample (window) to estimate std,\n",
    "#                    default = 500 (for whole genome screens)\n",
    "\n",
    "\n",
    "class Args:\n",
    "    infile = \"./sgRNA_count.txt\"\n",
    "    drugz_output_file = \"./drugz_results.txt\"\n",
    "    fc_outfile = \"./fc_results.txt\"\n",
    "    control_samples = \"T15_A_control,T15_B_control,T15_C_control\"\n",
    "    drug_samples = \"T15_A_olaparib,T15_B_olaparib,T15_C_olaparib\"\n",
    "    remove_genes = \"LacZ,luciferase,EGFR\"\n",
    "    unpaired = False\n",
    "    pseudocount = 5\n",
    "    half_window_size = 5  # 5 because of the size of test data set (sgRNA_count.txt = 9 guides (i.e. rows))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:drugz:Initiating analysis\n",
      "INFO:drugz:Loading the read count matrix\n",
      "INFO:drugz:Normalizing read counts\n",
      "INFO:drugz:Calculating raw fold change for replicate 1\n",
      "INFO:drugz:Caculating smoothed Empirical Bayes estimates of stdev for replicate 1\n",
      "INFO:drugz:Caculating guide-level Zscores for replicate 1\n",
      "INFO:drugz:Calculating raw fold change for replicate 2\n",
      "INFO:drugz:Caculating smoothed Empirical Bayes estimates of stdev for replicate 2\n",
      "INFO:drugz:Caculating guide-level Zscores for replicate 2\n",
      "INFO:drugz:Calculating raw fold change for replicate 3\n",
      "INFO:drugz:Caculating smoothed Empirical Bayes estimates of stdev for replicate 3\n",
      "INFO:drugz:Caculating guide-level Zscores for replicate 3\n",
      "INFO:drugz:Caculating gene-level Zscores\n",
      "INFO:drugz:Writing output file paired results\n"
     ]
    }
   ],
   "source": [
    "drugz_results = dz.drugZ_analysis(Args())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sumZ</th>\n",
       "      <th>numObs</th>\n",
       "      <th>normZ</th>\n",
       "      <th>pval_synth</th>\n",
       "      <th>rank_synth</th>\n",
       "      <th>fdr_synth</th>\n",
       "      <th>pval_supp</th>\n",
       "      <th>rank_supp</th>\n",
       "      <th>fdr_supp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENE</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>1.640558</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.158655</td>\n",
       "      <td>1</td>\n",
       "      <td>0.317311</td>\n",
       "      <td>0.841345</td>\n",
       "      <td>2</td>\n",
       "      <td>0.841345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>3.095789</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.841345</td>\n",
       "      <td>2</td>\n",
       "      <td>0.841345</td>\n",
       "      <td>0.158655</td>\n",
       "      <td>1</td>\n",
       "      <td>0.317311</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          sumZ  numObs  normZ  pval_synth  rank_synth  fdr_synth  pval_supp  \\\n",
       "GENE                                                                          \n",
       "A1CF  1.640558     9.0   -1.0    0.158655           1   0.317311   0.841345   \n",
       "A1BG  3.095789    18.0    1.0    0.841345           2   0.841345   0.158655   \n",
       "\n",
       "      rank_supp  fdr_supp  \n",
       "GENE                       \n",
       "A1CF          2  0.841345  \n",
       "A1BG          1  0.317311  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drugz_results\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Option 2. Run the analysis as a program and specify the output file (e.g. drugZ_test.txt)¶\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run drugz.py -i sgRNA_count.txt -o drugZ_results.txt -c T15_A_control,T15_B_control,T15_C_control -x T15_A_olaparib,T15_B_olaparib,T15_C_olaparib --half_window_size 5"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "1b992bd963af861f18abdd13ee61f91cbc1bd0838f7cc1ddaa99fcd03e5d0c18"
  },
  "kernelspec": {
   "display_name": "Python 3.9.9 64-bit ('bio': pyenv)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.9"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
